web/client/utils/CoordinatesUtils.js

```diff
@@ -392,6 +395,49 @@ const CoordinatesUtils = {
 
         return azimuth;
     },
+    /**
+    @param coordinates in EPSG:4326
+    return vincenty distance between two points
+    */
+    calculateVincentyDistance: (coordinates) => {
+        let length = 0;
+        for (let i = 0; i < coordinates.length - 1; ++i) {
+            const p1 = coordinates[i];
+            const p2 = coordinates[i + 1];
+            const [x1, y1] = p1;
+            const [x2, y2] = p2;
+            length += parseFloat(geo.vincentySync({longitude: x1, latitude: y1}, {longitude: x2, latitude: y2}));
+        }
+        return length;
+    },
+    /**
+    @param coordinates in EPSG:4326
+    return distance between two points using haversine formula
+    */
+    calculateGeodesicDistance: (coordinates) => {
```

Comment:
const formulas = {
   geodesic: (coords) => ...,
   vincenty: (coords) => ...
};
calculateDistance: (coordinates, formula = 'geodesic') => { return formulas[formula](coordinates); }

Reply:
ok, if i do that i have to change also the reducer
do you also mean to change the formula value from "Haversine" to "Geodesic" ?

Reply:
The important thing is: replace if else with a list of formulas where I can add new ones in the future. the names are not important

Reply:
that's right, i see your point